### ISSUE TITLE
Updating to Docker Desktop 2.2.0.0 breaks Docksal

### ISSUE THREAD

**Issue by ygerasimov** [docksal/docksal#1268]
I have updated my Docker Desktop to 2.2.0 and can't start docksal anymore. When I run `fin up` I am getting an error
```
Resetting Docksal system services...
 * proxy
docker: Error response from daemon: driver failed programming external connectivity on endpoint docksal-vhost-proxy (83b3ef7bafa2d27d0af784b64c635e1fff31b26803dad7bca9a4b20eb98457fb): Error starting userland proxy: listen tcp 192.168.64.100:8443: bind: cannot assign requested address.
 ERROR:  Failed starting the proxy service.
```

I did try to change ports but didn't help. My docksal version is
```
ygerasimov$ fin --version
fin version: 1.93.0
```

**Comment by achekulaev** [comment]
Right now the solution is to downgrade

**Comment by lmakarov** [comment]
Here's the direct link to the latest 2.1.x version (2.1.0.5):
https://download.docker.com/mac/stable/40693/Docker.dmg

**Comment by lmakarov** [comment]
Upstream issue: https://github.com/docker/for-mac/issues/4209

**Comment by kenrbnsn** [comment]
One gets the same error with the Windows 10 version of Docker 2.2.0.0

**Comment by tekNorah** [comment]
I am also experiencing the same issue (Mac).

**Comment by 42robotsCS** [comment]
I can confirm this bug when upgrading to Docker Desktop 2.2.0.0 resulting in: 

`Error starting userland proxy: listen tcp 192.168.64.100:443: bind: cannot assign requested address.`

Going back to Docker Desktop Version 2.1.0.5 resolves the issue for now.

**Comment by rabithk** [comment]
+1

**Comment by cmodern** [comment]
+1  2

**Comment by uniquename** [comment]
+1

**Comment by akosglue** [comment]
+1 4

**Comment by achekulaev** [comment]
I do not see any point of adding "+1" comments here. It is an upstream issue. I urge you to go and leave your comments there https://github.com/docker/for-mac/issues/4209

**Comment by nerfchicken** [comment]
While it may be an upstream issue, I don't think it impacts only docker/for-mac. The same issue impacted all of our Windows users when they upgraded to 2.2.0

**Comment by tekNorah** [comment]
> While it may be an upstream issue, I don't think it impacts only docker/for-mac. The same issue impacted all of our Windows users when they upgraded to 2.2.0

You are correct, it is also an issue for windows.  See: https://github.com/docker/for-win/issues/5546

**Comment by danshumaker** [comment]
It also seems strange to upvote a "solution" that is a downgrade.  It's ok that we can all downgrade to get around the issue, but it should not be considered a good solution to downgrade.  If anything a downgrade "solution" should get a thumbs down.  ?? anyway no big deal about emoji's/thumbs/votes.

I just confirmed that this also happens for docker for mac version 2.2.0.3 .and fin version 1.93.0

**Comment by lmakarov** [comment]
> It also seems strange to upvote a "solution" that is a downgrade. It's ok that we can all downgrade to get around the issue, but it should not be considered a good solution to downgrade. If anything a downgrade "solution" should get a thumbs down. ??

@danshumaker Downgrading is a temporary workaround. The actual issue is [upstream](https://github.com/docksal/docksal/issues/1268#issuecomment-577038487) in Docker for Mac/Win. Chiming in there will make the most sense.

**Comment by phox4ever** [comment]
@lmakarov Isn't Docksal using the DockeNat interface? It has been removed: docker/for-win/#5538

**Comment by lmakarov** [comment]
> Isn't Docksal using the vEthernet (DockerNAT) interface? It has been removed: docker/for-win#5538

@phox4ever oh wow... That's an unfortunate change and will definitely require refactoring in Docksal.
At the same time, it is a different (and additional) compatibility issue with the new Docker Desktop for Windows versions.

That's for bringing this up. I submitted a separate issue here: https://github.com/docksal/docksal/issues/1276

**Comment by hugronaphor** [comment]
Would be nice to mention a specific supported version on https://docs.docksal.io/getting-started/setup/#install-macos as it took me a while to get to this discussion.

**Comment by kenrbnsn** [comment]
Has this been resolved yet? Upgrading to the latest  Docker Desktop on Windows still breaks docksal.

![image](https://user-images.githubusercontent.com/43186/78252878-8493a480-74c1-11ea-962a-7db8bee1053e.png)

**Comment by lpeabody** [comment]
@kenrbnsn it's an upstream issue with Docker Desktop. In the mac issue queue they said it's fixed in an upcoming edge release https://github.com/docker/for-mac/issues/4209#issuecomment-605991683. Don't know if there is a corresponding release for Windows, however.

**Comment by lmakarov** [comment]
The upstream edge releases with the fix are now available for download - v2.2.3.0 (edge build v43965).

[Mac](https://docs.docker.com/docker-for-mac/edge-release-notes/#docker-desktop-community-2230): https://download.docker.com/mac/edge/43965/Docker.dmg
[Windows](https://docs.docker.com/docker-for-windows/edge-release-notes/#docker-desktop-community-2230): https://download.docker.com/win/edge/43965/Docker%20Desktop%20Installer.exe

Please try and report your experience.
I would still recommend waiting for a stable Docker Desktop release for most users.

**Comment by jacobsaw** [comment]
Confirmed that this issue is resolved after upgrading to edge 2.2.3.0.

**Comment by kenrbnsn** [comment]
Still not working for me.

Docker:
![image](https://user-images.githubusercontent.com/43186/78605474-965eaa00-7829-11ea-9724-82979e738e72.png)

Docksal:
$ fin version
fin version: 1.93.0
$ fin start
 ALERT:  Some Docksal services were not running
         Restarting Docksal system services...
Configuring network settings...
Resetting Docksal system services...
 * proxy
docker: Error response from daemon: Ports are not available: listen tcp 192.168.64.100:443: bind: The requested address is not valid in its context.
 ERROR:  Failed starting the proxy service.

**Comment by lmakarov** [comment]
> Still not working for me.

@kenrbnsn you are on Windows, correct?

**Comment by kenrbnsn** [comment]
@lmakarov Yes. Windows 10
![image](https://user-images.githubusercontent.com/43186/78915109-6e558d80-7a59-11ea-8713-988091cdf3ed.png)

**Comment by ovangmbh** [comment]
On my mac with catalina (10.15.4) it works with edge 2.2.3.0 , but I have to run always "fin update" after "fin project start", because the dns to "whatever.docksal" is not working.

**Comment by thejoshsmith** [comment]
Latest stable version of Docker for Mac (2.3.0.2) is working.

**Comment by lmakarov** [comment]
The latest stable Docker Desktop versions (released today) should address this regression:

- [Docker for Mac 2.3.0.2](https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-community-2302) ([download](https://download.docker.com/mac/stable/45183/Docker.dmg)) - **CONFIRMED**
> Fixed an issue where a container port could not be exposed on a specific host IP. See docker/for-mac#4209.
- [Docker for Windows 2.3.0.2](https://docs.docker.com/docker-for-windows/release-notes/#docker-desktop-community-2302) ([download](https://download.docker.com/win/stable/45183/Docker%20Desktop%20Installer.exe))
> Fixed an issue where a container port could not be exposed on a specific host IP. See docker/for-win#5546.

**Comment by mike-potter** [comment]
Just tried Docker for Mac 2.3.0.2 and there still seems to be a problem.  I just get "The connection was reset" when trying to visit my http://project.docksal site.  I can "ping" the site, but doing a `curl http://www.project.docksal` returns no data.

As mentioned in the comment above, doing a "fin system reset" after doing the "fin project start" fixed the problem.

Looks like maybe the initial "fin start" doesn't work properly after doing the Docker Desktop update.  But once you do a "fin system reset" (which *looks* like it is doing the same stuff as "fin start") then everything starts working.

**Comment by mike-potter** [comment]
Also confirming that if you restart Docker Desktop and then do a `fin start`, the project site no longer responds.  Still have to do a `fin system reset` to get the site to work again.  Once that is done, doing `fin stop` and `fin start` seem to work fine.

So you just need to use `fin system reset` after rebooting or restarting Docker Desktop.  Hopefully this is something that can be fixed on the Docksal side.  The bug that prevented Docker Desktop 2.2 from working *DOES* seem to be resolved.  Needing to use `fin system reset` is a new issue.

**Comment by mike-potter** [comment]
Opened new issue #1342 to track this.

**Comment by 42robotsCS** [comment]
Can confirm this, see  comment in #1342.
Not sure if the need of using 

`fin system reset`

 is an issue at all ? But i cannot judge that, just interested.

**Comment by lmakarov** [comment]
Should be fixed in [v1.13.3](https://github.com/docksal/docksal/releases/tag/1.13.3) (just released).
